src/gitProviders/github.tsx

```diff
@@ -0,0 +1,222 @@
+        try {
+            await this.authenticate(this.props.context).post(
+                `${this.props.context.baseUrl}/repos/${this.props.context.organization}/${this.props.context.projectId}/contents/${this.props.context.configFileName}`,
+                {
+                    "branch": this.props.context.branch,
+                    "content": "IA==",
+                    "message": `Init new config file ${this.props.context.configFileName}`
+                }
+            );
+        } catch(e) {
+            console.error(e.response);
+            throw 'Creating a new file via Github API failed.'
+        }
+    }
+
+    async createRemoteBranchFromCurrent(branchName) {
+        try {
+            await this.authenticate(this.props.context).post(`${this.props.context.baseUrl}/repos/${this.props.context.organization}/${this.props.context.projectId}/git/refs`,{"ref":"refs/heads/"+branchName,"sha":this.props.context.branch});
+        } catch(e) {
+            console.error(e.response);
+            throw 'Creating a new branch via Github API failed.'
+        }
+    }
+
+    async fetchBranches() {
+        if (!this.props?.context.baseUrl || !this.props?.context.projectId || !this.props?.context.token) {
+            return [];
+        }
+        try {
+            const response = await this.authenticate(this.props.context).get(`${this.props.context.baseUrl}/repos/${this.props.context.organization}/${this.props.context.projectId}/git/refs/heads`,);
+            return response.data.map((o) => {
+                let heads = o.ref.split("/");
+                return heads[heads.length-1];
+            });
+
+        } catch(e) {
+            console.error(e);
+            throw 'Fetching the projects branches via Github API failed.'
+        }
+    }
+
+    async pullWorkspace(context) {
+        try {
+            const getFileResponse = await this.authenticate(context).get(`${context.baseUrl}/repos/${context.organization}/${context.projectId}/contents/${context.configFileName}`);
+            const response = await this.authenticate(context).get(getFileResponse.data.download_url);
+            return(response.data);
+        } catch (e) {
+            console.error(e);
+            throw 'Fetching the workspace via Github API failed.'
+        }
+    }
+
+    findObjectByKey(array, key, value) {
+        for (var i = 0; i < array.length; i++) {
+            if (array[i][key] === value) {
+                return array[i];
+            }
+        }
+        return null;
+    }
+
+    async pushWorkspace(content, messageCommit,context) {
+        try {
+            const buff = Buffer.from(content, 'utf-8');
+            const contentBase64 = buff.toString('base64');
+
+            let data = {
+                "branch": context.branch,
+                "message": messageCommit,
+                "content":contentBase64,
+                sha: null
+            };
+            console.log()
+            console.log(this.props);
```

Comment:
Maybe log with short description or keyword beforehand.
We could start a convention and log to .debug() instead of general stdout with e.g.:
```
console.debug("props:", this.props);
```